templates/kubernetes/terraform/modules/kubernetes/irsa.tf

```diff
@@ -0,0 +1,48 @@
+# IRSA support: allow application specific policy via service-account and role
+
+# application_policy_list is passed from main.tf as below:
+  # application_policy_list = [
+  #   {
+  #     service_account = "app1"
+  #     namespace       = "piggycloud-me"
+  #     policy          = data.aws_iam_policy_document.resource_access_app1
+  #   },
+  #   {
+  #     service_account = "app2"
+  #     namespace       = "piggycloud-me"
+  #     policy          = data.aws_iam_policy_document.resource_access_app2
+  #   }
+  #   # could be more here
+  # ]
+
+# Create a role using oidc to map service accounts
+module "iam_assumable_role_irsa" {
+  count                         = length(var.application_policy_list)
+  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version                       = "~> v2.14.0"
+  create_role                   = true
+  role_name                     = "${var.project}-k8s-${var.environment}-${var.application_policy_list[count.index].service_account}"
+  provider_url                  = replace(data.aws_eks_cluster.cluster.identity.0.oidc.0.issuer, "https://", "")
+  role_policy_arns              = [aws_iam_policy.irsa[count.index].arn]
+  oidc_fully_qualified_subjects = ["system:serviceaccount:${var.application_policy_list[count.index].namespace}:${var.application_policy_list[count.index].service_account}"]
+}
+
+# Create policies
+resource "aws_iam_policy" "irsa" {
+  count       = length(var.application_policy_list)
+  name_prefix = var.project
```

Comment:
Should probably be project + environment in case someone has two environments in a single aws account.

Reply:
done